### PR TITLE
state computer working with block metadata ext

### PR DESCRIPTION
**Pull request by zjma**
### Description

When randomness feature is enabled, execute `BlockMetadataExt` flow.

### CODE REVIEW

File: consensus/consensus-types/src/executed_block.rs [hunk 1]
```diff
@@ -157,27 +153,6 @@ impl ExecutedBlock {
         )
     }
 
-    pub fn transactions_to_commit(
```
Comment (zjma) [inline consensus/consensus-types/src/executed_block.rs:160]: this is no longer a matter of a block, but a state_computer thing now.

**Review by danielxiangzl — COMMENTED** [review]
How do you test this PR?

**Comment by zjma** [comment]
> How do you test this PR?

will adapt `consensus/src/state_computer_tests.rs` for this.

**Comment by zjma** [comment]
> How do you test this PR?

also i guess everything starting dkg_5 should be rebased onto #11458

**Review by danielxiangzl — APPROVED** [review]

**Review by grao1991 — APPROVED** [review]

**Pull request merged**